src/Components/SearchLocationInput/SearchLocationInput.js

```diff
@@ -0,0 +1,80 @@
+import React, { useState, useEffect, useRef } from "react";
+let autoComplete;
+
+const loadScript = (url, callback) => {
+  let script = document.createElement("script");
+  script.type = "text/javascript";
+
+
+  if (script.readyState) {
+    script.onreadystatechange = function() {
+      if (script.readyState === "loaded" || script.readyState === "complete") {
+        script.onreadystatechange = null;
+        callback();
+      }
+    };
+  } else {
+    script.onload = () => callback();
+  }
+
+  script.src = url;
+  document.getElementsByTagName("head")[0].appendChild(script);
+};
+
+
+function handleScriptLoad(updateQuery, autoCompleteRef) {
+
+  autoComplete = new window.google.maps.places.Autocomplete(
+    autoCompleteRef.current,
+  // specify location type below, in 'types' array
+    { types: ["address"], componentRestrictions: { country: "us" } }
+  );
+
+  autoComplete.setFields(["address_components", "formatted_address"]);
+
+  autoComplete.addListener("place_changed", () =>
+    handlePlaceSelect(updateQuery)
+  );
+}
+
+async function handlePlaceSelect(updateQuery) {
+  const addressObject = autoComplete.getPlace();
+  const query = addressObject.formatted_address;
+  updateQuery(query);
+  console.log(addressObject);
+}
+
+function SearchLocationInput() {
+  const [query, setQuery] = useState("");
+  const autoCompleteRef = useRef(null);
+
+  useEffect(() => {
+    loadScript(
+      `https://maps.googleapis.com/maps/api/js?key=${process.env.REACT_APP_AUTOCOMPLETE_API_KEY}&libraries=places`,
+      () => handleScriptLoad(setQuery, autoCompleteRef)
+    );
+  }, []);
+
+  return (
+    <div className="search-location-section">
+      <input
+        ref={autoCompleteRef}
+        onChange={event => setQuery(event.target.value)}
+        placeholder="Starting address"
+        value={query}
+        className='location-input'
+        required
+      />
+      <input
+        // ref={autoCompleteRef}
```

Comment:
Going to look into this tonight and see what I find. I'll circle back once I know more.

Reply:
Sounds good - I'll do the same tomorrow.